extensions/io/org.eclipse.smarthome.io.iota/src/main/java/org/eclipse/smarthome/io/iota/utils/IotaUtilsImpl.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.iota.utils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import jota.IotaAPI;
+import jota.dto.response.GetBalancesResponse;
+import jota.dto.response.SendTransferResponse;
+import jota.error.ArgumentException;
+import jota.model.Transfer;
+import jota.utils.InputValidator;
+import jota.utils.TrytesConverter;
+
+/**
+ * The {@link IotaUtilsImpl} provides utils methods to work with IOTA transactions.
+ *
+ * @author Theo Giovanna - Initial Contribution
+ */
+public class IotaUtilsImpl implements IotaUtils {
+
+    private final Logger logger = LoggerFactory.getLogger(IotaUtilsImpl.class);
+    private static final String PATH = "../../extensions/io/org.eclipse.smarthome.io.iota/lib/mam/example/";
+    private String seed = null;
+    private int start = 0;
+    private IotaAPI iotaAPI;
+    private Process process;
+    private String oldResult = "";
+    private final String npmPath = isWindows() ? "npm.cmd" : "/usr/local/bin/node";
```

Comment:
Unfortunately this is still an educated guess. Can this be done in the IotaIo config? Then there must be a way to pass it to the IOTABridgeHandler.
Another way would be to have the IotaIo service injected into the binding's handler factory and use it as "the" communication object. This way the one configuration and one service would do all the Tangle communication. wdyt?